code/drasil-lang/lib/Language/Drasil/Expr/Lang.hs

```diff
@@ -160,6 +160,38 @@ data Expr where
   Operator :: AssocArithOper -> DiscreteDomainDesc Expr Expr -> Expr -> Expr
   -- | A different kind of 'IsIn'. A 'UID' is an element of an interval.
   RealI    :: UID -> RealInterval Expr Expr -> Expr
+  -- | A clif is an expression
+  Clif     :: Clif -> Expr
+
+-- | The basis in which to project clifs
+data Basis where
+  -- | ℝⁿ
+  Rn :: Int -> Basis
```

Comment:
I think we have the opportunity here to use `Natural` or `Nat` from `Data.Nat`. Let's build more invariants into our representations!